packages/babel-parser/test/fixtures/es2015/modules/export-declaration-trailing-comma/input.js

```diff
@@ -0,0 +1,5 @@
+export {
+  /* One */ Foo
+  /* Two */,
+  /* Three */
+} /* Four */ from "foo";
```

Comment:
Shouldn't "Four" be an `innerComment` of `ExportNamedDeclaration`?

Reply:
That would be better, but from what I can tell, there's no way to figure out which comments are before the closing bracket and which ones are after. Before the PR, the last three comments get attached as `leadingComments` of `foo`, so we were at least not dropping them. So I could remove handling of `ImportDeclaration` and `ExportDeclaration` from this PR if you prefer the prior logic.

Reply:
I tried to call `adjustCommentsAfterTrailingComma` right before eating the `}` in `parseNamedImportSpecifiers`, instead of doing so in `processComment`. By doing so, `Four` gets attached as a `leadingComments` of `"foo"`.

I'm ok with either behaviors (`leadingComments` of `"foo"` or `trailingComments` of `Foo`); attaching it as an `innerComment` is too hard with the current comments attachment logic.

Reply:
Agree. Comment attachment currently could only be invoked at `finishNode`, and it will definitely misclassify some inner comments as long as there are multiple whitespaces between the `end` location of any two nodes.

Reply:
Yeah, looks like something like #9084 is needed to overhaul comment parsing properly, but there might be challenges around backwards compatibility.

I'm slightly more inclined to keep them as `trailingComments` on `Foo`, because I expect there to be more examples of trailing comments before the `}` than after, so I'll leave this in.